liesel/goose/builder.py

```diff
@@ -136,7 +136,75 @@ def set_initial_values(self, model_state: ModelState, multiple_chains=False):
         self._model_state = Option(model_states)
 
     def set_jitter_fns(self, jitter_fns: JitterFunctions | None):
-        """Set the jittering functions."""
+        """
+        Set the jittering functions.
+
+        A jittering function is a function that takes as input a key and a value,
+        and applies a random jittering (noise) to the input value
+        based on the given key.
+        This function takes as input a dictionary where a jittering function is
+        assigned to a position key.
+
+        Examples
+        --------
```

Comment:
As a general point, I think an extensive example greatly benefits from some narrative comments in between the code blocks, commenting about what happens in those code blocks.  That makes it much easier to follow and see where the most important parts are.